### PR TITLE
Add e2e test for modifying clientparams

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #2120

### Commit Message / Changelog Entry

```bash
e2e: add e2e test for modifying client params
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/core/02-client/client_test.go [hunk 1]
```diff
@@ -242,6 +253,57 @@ func (s *ClientTestSuite) TestClient_Update_Misbehaviour() {
...
+			s.ExecuteGovProposal(ctx, chainA, chainAWallet, proposal)
+		}
+	})
+
+	t.Run("validate the param was successfully changed", func(t *testing.T) {
+		allowedClients := s.QueryAllowedClients(ctx, chainA)
+		s.Require().Equal([]string{allowedClient}, allowedClients)
+	})
+
+	t.Run("ensure querying non-allowed client's status returns Unauthorized Status", func(t *testing.T) {
```
Comment (DimitrisJim) [inline e2e/tests/core/02-client/client_test.go:300]: hits the same check we'd like only in a different part of 02-client, can change if required (to try and actually create a client and hit the check in `CreateClient`.

File: e2e/tests/core/02-client/client_test.go [hunk 2]
```diff
@@ -242,6 +253,57 @@ func (s *ClientTestSuite) TestClient_Update_Misbehaviour() {
...
+			}
+
+			proposal := paramsproposaltypes.NewParameterChangeProposal(ibctesting.Title, ibctesting.Description, changes)
+			s.ExecuteGovProposal(ctx, chainA, chainAWallet, proposal)
+		}
+	})
+
+	t.Run("validate the param was successfully changed", func(t *testing.T) {
+		allowedClients := s.QueryAllowedClients(ctx, chainA)
+		s.Require().Equal([]string{allowedClient}, allowedClients)
```
Comment (damiannolan) [inline e2e/tests/core/02-client/client_test.go:297]: nit: we could maybe use `s.Require().Contains()` here, makes no difference to me

File: e2e/tests/core/02-client/client_test.go [hunk 3]
```diff
@@ -242,6 +253,57 @@ func (s *ClientTestSuite) TestClient_Update_Misbehaviour() {
...
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 1, chainA, chainB), "failed to wait for blocks")
+
+	t.Run("ensure allowed clients are set to the default", func(t *testing.T) {
+		allowedClients := s.QueryAllowedClients(ctx, chainA)
+		s.Require().Equal(clienttypes.DefaultAllowedClients, allowedClients)
+	})
+
+	allowedClient := ibcexported.Solomachine
+	t.Run("change the allowed client to only allow localhost clients", func(t *testing.T) {
```
Comment (damiannolan) [inline e2e/tests/core/02-client/client_test.go:275]: should this be solomachine clients? 

```suggestion
	allowedClient := ibcexported.Solomachine
	t.Run("change the allowed client to only allow solomachine clients", func(t *testing.T) {
```

**Review by damiannolan — APPROVED** [review]
Awesome, thanks @DimitrisJim 🚀

**Review by chatton — APPROVED** [review]
LGTM test looks great!

Maybe we can add it to the relevant [compatibility matrices](https://github.com/cosmos/ibc-go/blob/678197af1ceb3f379359f313cfea7b1b38a35315/.github/compatibility-test-matrices/release-v7.1.x/client-chain-a.json#L22)

**Pull request merged**